test/Common.cpp

```diff
@@ -91,22 +91,27 @@ CommonOptions::CommonOptions(std::string _caption):
 		po::options_description::m_default_line_length,
 		po::options_description::m_default_line_length - 23
 	)
+{
+
+}
+
+void CommonOptions::addOptions()
 {
 	options.add_options()
 		("evm-version", po::value(&evmVersionString), "which evm version to use")
 		("testpath", po::value<fs::path>(&this->testPath)->default_value(solidity::test::testPath()), "path to test files")
 		("vm", po::value<std::vector<fs::path>>(&vmPaths), "path to evmc library, can be supplied multiple times.")
-		("ewasm", po::bool_switch(&ewasm), "tries to automatically find an ewasm vm and enable ewasm test-execution.")
-		("no-semantic-tests", po::bool_switch(&disableSemanticTests), "disable semantic tests")
-		("no-smt", po::bool_switch(&disableSMT), "disable SMT checker")
-		("optimize", po::bool_switch(&optimize), "enables optimization")
-		("enforce-via-yul", po::bool_switch(&enforceViaYul), "Enforce compiling all tests via yul to see if additional tests can be activated.")
-		("enforce-compile-to-ewasm", po::bool_switch(&enforceCompileToEwasm), "Enforce compiling all tests to Ewasm to see if additional tests can be activated.")
-		("enforce-gas-cost", po::bool_switch(&enforceGasTest), "Enforce checking gas cost in semantic tests.")
-		("enforce-gas-cost-min-value", po::value(&enforceGasTestMinValue), "Threshold value to enforce adding gas checks to a test.")
-		("abiencoderv1", po::bool_switch(&useABIEncoderV1), "enables abi encoder v1")
-		("show-messages", po::bool_switch(&showMessages), "enables message output")
-		("show-metadata", po::bool_switch(&showMetadata), "enables metadata output");
+		("ewasm", po::bool_switch(&ewasm)->default_value(ewasm), "tries to automatically find an ewasm vm and enable ewasm test-execution.")
+		("no-semantic-tests", po::bool_switch(&disableSemanticTests)->default_value(disableSemanticTests), "disable semantic tests")
+		("no-smt", po::bool_switch(&disableSMT)->default_value(disableSMT), "disable SMT checker")
+		("optimize", po::bool_switch(&optimize)->default_value(optimize), "enables optimization")
+		("enforce-via-yul", po::value<bool>(&enforceViaYul)->default_value(enforceViaYul)->implicit_value(true), "Enforce compiling all tests via yul to see if additional tests can be activated.")
+		("enforce-compile-to-ewasm", po::bool_switch(&enforceCompileToEwasm)->default_value(enforceCompileToEwasm), "Enforce compiling all tests to Ewasm to see if additional tests can be activated.")
+		("enforce-gas-cost", po::value<bool>(&enforceGasTest)->default_value(enforceGasTest)->implicit_value(true), "Enforce checking gas cost in semantic tests.")
```

Comment:
These options show up like this in the command-line help now:
```
  --enforce-via-yul [=arg(=1)] (=1)
  --enforce-gas-cost [=arg(=1)] (=1)
                       Enforce checking gas cost in semantic tests.
  --enforce-gas-cost-min-value arg (=100000)
                       Threshold value to enforce adding gas checks to a test.
```

Is there a way to show something like `--enforce-via-yul [=arg(=true)] (=true)` (or even better `--enforce-via-yul (=true)`) instead? If not, I guess we can live with that but it being `1` makes it look a bit less like a simple switch and more like an option that actually requires some specific integer value.

Reply:
That is internal to the `program_options` implementation, unfortunately.